clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -354,6 +379,13 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                             (shard->skipped_instrs_ && shard->stream != nullptr &&
                              shard->stream->get_page_size() > 0),
                         "Missing page size marker");
+        report_if_false(
+            shard,
+            (shard->found_syscall_marker_ &&
```

Comment:
Use `shard->found_syscall_marker_ == TESTANY(OFFLINE_FILE_TYPE_SYSCALL_NUMBERS, shard->file_type_)`?

Reply:
Maybe because of my C background I am wary of equality of booleans which is not safe in C.  In C++ it is much safer and here if we assume that TESTANY is in fact a bool type (ends with a `!=`) it seems like we have two bool types and the C++ comparison is safe: but we could have defined TESTANY as just a bitwise `&` and then we have to be confident in which types will be converted to what.  If the bool is converted to an int then this is not safe.

How about I use the == but I explicitly static_cast the TESTANY to a bool -- that makes me more comfortable.